command/bridge/deploy/deploy.go

```diff
@@ -604,73 +295,69 @@ func deployContracts(outputter command.OutputFormatter, client *jsonrpc.EthClien
 		})
 	}
 
-	if err := g.Wait(); err != nil {
-		return collectResultsOnError(results), err
-	}
-
-	commandResults := make([]command.CommandResult, 0, len(results))
-
-	for _, result := range results {
-		commandResults = append(commandResults, result)
-
-		populatorFn, exists := metadataPopulatorMap[result.Name]
-		if !exists {
-			continue
-		}
-
-		populatorFn(rootchainConfig, result.Address)
-	}
-
-	g, ctx = errgroup.WithContext(cmdCtx)
-
-	for contractName := range results {
-		contractName := contractName
-
-		initializer, exists := initializersMap[contractName]
-		if !exists {
-			continue
+	initializeContractFn := func(contract *contract, txr txrelayer.TxRelayer, chainID int64) {
+		if contract.initializeFn == nil {
+			// some contracts do not have initialize function
+			return
 		}
 
 		g.Go(func() error {
 			select {
-			case <-cmdCtx.Done():
-				return cmdCtx.Err()
+			case <-ctx.Done():
+				return ctx.Err()
 			default:
-				return initializer(outputter, txRelayer, initialValidators, rootchainConfig, deployerKey, chainID)
+				return contract.initializeFn(outputter, externalTxRelayer,
+					initialValidators, bridgeConfig, deployerKey, chainID)
+
 			}
 		})
 	}
 
+	// deploy external contracts
+	for _, contract := range externalContracts {
+		deployContractFn(contract, externalTxRelayer)
+	}
+
+	// deploy internal contracts if blade is live
+	if !params.isBootstrap {
+		for _, contract := range internalContracts {
+			deployContractFn(contract, internalTxRelayer)
+		}
+	}
+
+	// wait for all contracts to be deployed
 	if err := g.Wait(); err != nil {
-		return deploymentResultInfo{RootchainCfg: nil, CommandResults: nil}, err
+		return collectResultsOnError(results), err
 	}
 
-	return deploymentResultInfo{
-		RootchainCfg:   rootchainConfig,
-		CommandResults: commandResults}, nil
-}
+	// initialize contracts only after all of them are deployed
+	g, ctx = errgroup.WithContext(cmdCtx)
 
-// populateExistingTokenAddr checks whether given token is deployed on the provided address.
-// If it is, then its address is set to the rootchain config, otherwise an error is returned
-func populateExistingTokenAddr(eth *jsonrpc.EthClient, tokenAddr, tokenName string,
-	rootchainCfg *polybft.RootchainConfig) error {
-	addr := types.StringToAddress(tokenAddr)
+	// initialize external contracts
+	for _, contract := range externalContracts {
+		initializeContractFn(contract, externalTxRelayer, internalChainID)
+	}
 
-	code, err := eth.GetCode(addr, jsonrpc.LatestBlockNumberOrHash)
-	if err != nil {
-		return fmt.Errorf("failed to check is %s token deployed: %w", tokenName, err)
-	} else if code == "0x" {
-		return fmt.Errorf("%s token is not deployed on provided address %s", tokenName, tokenAddr)
+	// initialize internal contracts if blade is live
+	if !params.isBootstrap {
+		for _, contract := range internalContracts {
+			initializeContractFn(contract, internalTxRelayer, externalChainID.Int64())
+		}
 	}
```

Comment:
```suggestion
	// initialize internal contracts if blade is live
	for _, contract := range internalContracts {
		initializeContractFn(contract, internalTxRelayer, externalChainID.Int64())
	}
```

Reply:
We are only deploying and initializing internal contracts if blade is live (`isBootstrap` is false), if it is not live, we do the pre-allocate.